exampleCustomAudioDevice/src/main/java/com/twilio/examplecustomaudiodevice/CustomAudioDeviceActivity.kt

```diff
@@ -0,0 +1,428 @@
+package com.twilio.examplecustomaudiodevice
+
+import android.Manifest
+import android.content.Context
+import android.content.DialogInterface
+import android.content.pm.PackageManager
+import android.media.AudioAttributes
+import android.media.AudioFocusRequest
+import android.media.AudioManager
+import android.os.Build
+import android.os.Bundle
+import android.support.design.widget.FloatingActionButton
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import com.koushikdutta.ion.Ion
+import com.twilio.examplecustomaudiodevice.dialog.Dialog
+import com.twilio.video.*
+import java.util.*
+
+class CustomAudioDeviceActivity : AppCompatActivity() {
+    /*
+     * Access token used to connect. This field will be set either from the console generated token
+     * or the request to the token server.
+     */
+    private var accessToken: String? = null
+
+    /*
+     * A Room represents communication between a local participant and one or more participants.
+     */
+    private var room: Room? = null
+
+    /*
+     * Android application UI elements
+     */
+    private var localAudioTrack: LocalAudioTrack? = null
+    private var connectActionFab: FloatingActionButton? = null
+    private var inputSwitchFab: FloatingActionButton? = null
+    private var customAudioDeviceStatusText: TextView? = null
+    private var connectDialog: AlertDialog? = null
+    private var audioManager: AudioManager? = null
+    private var previousAudioMode = 0
+    private var previousMicrophoneMute = false
+    private var disconnectedFromOnDestroy = false
+    private val isSpeakerPhoneEnabled = true
+    var fileAndMicAudioDevice: FileAndMicAudioDevice? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
```

Comment:
The multi-line comments look a little off in this file with respect to spacing and alignment. Please ensure that multi line comments have a new line at the end.